sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -614,7 +614,11 @@ object ViewHelper extends SQLConfHelper with Logging {
     }.getOrElse(false)
     if (replace && uncache) {
       logDebug(s"Try to uncache ${name.quotedString} before replacing.")
-      checkCyclicViewReference(analyzedPlan, Seq(name), name)
```

Comment:
I don't think we want to support it officially. It's for backward compatibility only.

Reply:
I didn't get this. If we remove the cyclic check, how to detect a recursive view created by SQL.

Reply:
> I didn't get this. If we remove the cyclic check, how to detect a recursive view created by SQL.

It seems not necessary to check cyclic since we want to replace a view. The view in analyzedPlan is an old view, not the new view, even if they have the same name, thus I think maybe remove is better.

Reply:
> I don't think we want to support it officially. It's for backward compatibility only.

If we won't suport this, then why we change it for backward compatibility? It is not a valid behavior. What about leading people to use different name?

Reply:
> We don't support recursive view for view created by SQL DDL, but we DO support this for view created by dataset API.

Ah, I see, thanks for the explanation.